schedule_create.rb

```diff
@@ -122,19 +133,37 @@ def add_vm_params(vm_to_add_ref)
   end
 
   def vm_ref(virtual_server)
-    hierarchy_root_id =
-      uid_to_identifier(
-        :HierarchyRoot,
-        api_get(
-          build_query(:hierarchyroot, { uniqueid: "\"#{vcenter_instance_uuid(virtual_server)}\"" }, entities: false)
-        ).dig(:QueryResult, :Refs, :Ref, :UID)
-      )
-
-    return unless hierarchy_root_id
+    hierarchyroots =
+      api_get(
+        build_query(:hierarchyroot, { uniqueid: "\"#{vcenter_instance_uuid(virtual_server)}\"" }, entities: false)
+      ).dig(:QueryResult, :Refs, :Ref)
+
+    return unless hierarchyroots
```

Comment:
add method
`
def hierarchy_roots
  @hierarchy_roots ||= api_get()....
end
`

and then here simple

`return unless hierarchy_roots`

Reply:
Rejected.